openmmtools/multistate/multistatesampler.py

```diff
@@ -1309,45 +1309,73 @@ def _minimize_replica(self, replica_id, tolerance, max_iterations):
         thermodynamic_state_id = self._replica_thermodynamic_states[replica_id]
         thermodynamic_state = self._thermodynamic_states[thermodynamic_state_id]
         sampler_state = self._sampler_states[replica_id]
-
-        # Use the FIRE minimizer
-        integrator = FIREMinimizationIntegrator(tolerance=tolerance)
+        # NOTE: At this point, sampler_state.potential_energy is None. Initialize below.
+
+        logger.debug('Sampler state {}/{}:'.format(replica_id + 1, self.n_replicas))
+
+        # Temporarily disable the barostat during minimization by setting the
+        # pressure to None. (Otherwise, the minimizer will modify the box
+        # vectors and may cause instabilities which are very difficult to debug.)
+        pressure = thermodynamic_state.pressure
+        thermodynamic_state.pressure = None
+
+        # Use Gradient Descent first for numerical stability
+        integrator_grad = GradientDescentMinimizationIntegrator()
+        context_grad = thermodynamic_state.create_context(integrator_grad)
```

Comment:
It's good that we're creating the context and then cleaning it up again, since we don't repeatedly re-use the minimization context, but it would be useful if we also specified [platform properties to ensure we use mixed precision if needed](https://github.com/choderalab/openmmtools/blob/main/openmmtools/states.py#L1094) for the propagation or energy computation context caches.

Reply:
Done